gui/src/components/LogsViewer.svelte

```diff
@@ -81,26 +88,36 @@
       logViewport.scrollTop = logViewport.scrollHeight;
     }
   });
+
+  const setFilterStrDebounced = debounce((filterStr: string) => {
+    setFilterStr(workspaceId, workspace, filterStr);
+  }, 250);
 </script>
 
 <section>
   <h1>Logs</h1>
   {#if hasData(state.events)}
-    <div class="log-table-overflow-wrapper">
-      <div class="log-table-container" bind:this={logViewport}>
-        <table>
-          {#each state.events.data as event (event.id)}
-            <tr class="log-entry" style={logStyleFromHash(event.log)}>
-              <td>{shortDate(event.timestamp)}</td>
-              <td>{friendlyName(event.log)}</td>
-              <td>
-                <FormattedLogMessage message={event.message} />
-              </td>
-            </tr>
-          {/each}
-        </table>
+    <div class="log-overflow-wrapper">
+      <div class="log-container" bind:this={logViewport}>
+        {#each state.events.data as event (event.id)}
+          <div class="log-entry" style={logStyleFromHash(event.log)}>
+            <div class="timestamp">{shortDate(event.timestamp)}</div>
+            <div class="log">{friendlyName(event.log)}</div>
+            <div class="message">
+              <FormattedLogMessage message={event.message} />
+            </div>
+          </div>
+        {/each}
       </div>
     </div>
+    <input
+      placeholder="Filter..."
+      value={state.filterStr || ''}
```

Comment:
Does this work like a controlled input in React? I'm shocked that this seems to condense
```
value={stateX}
onChange={(e) => setStateX(e.currentTarget.value)}
```
into the far more concise
```
value={state.X || ''}
```
if so. Pretty cool, but it feels unintuitive that we don't have to explicitly call a change to the state.

Reply:
`value` here works like `defaultValue` in React, which is fine in this case because I just care about updating the external state, not synchronizing when someone else changes that state. To do the 2-way binding you are talking about is even simpler:

```
<script>
let x = '';
</script>

<input bind:value={x} />
```